api/clever/__init__.py

```diff
@@ -171,9 +171,64 @@ def _remote_exchange_payload(self, _db, code):
         )
 
     def remote_patron_lookup(self, token):
-        """Use a bearer token to look up detailed patron information.
+        """Use a bearer token for a patron to look up that patron's Clever
+        record through the Clever API.
+
+        This is the only method that has access to a patron's personal
+        information as provided by Clever. Here's an inventory of the
+        information we process and what happens to it:
+
+        * The Clever 'id' associated with this patron is passed out of
+          this method through the PatronData object, and persisted to
+          two database fields: 'patrons.external_identifier' and
+          'patrons.authorization_identifier'.
+
+          As far as we know, the Clever ID is an opaque reference
+          which uniquely identifies a given patron but contains no
+          personal information about them.
+
+        * The patron's personal name is passed out of this method
+          through the PatronData object. This information is available
+          for the duration of the active HTTP request (e.g. it could
+          be used to display a welcome message) but it cannot be
+          persisted to the database, because there's no place to store
+          it. You can verify this by looking at the PatronData class
+          in authenticator.py and the Patron class in
+          core/model/patron.py.
```

Comment:
Don't know if it needs to be mentioned here, but I believe the name is persisted by the mobile apps so they can show who is logged in.

Reply:
Hmm, I don't know how that would work. @kyles-ep Do you have access to the personal name of a patron who signed in to Open eBooks through Clever?

Reply:
The server-side part is https://github.com/NYPL-Simplified/circulation/blob/master/api/authenticator.py#L2466

Reply:
Hmm, OK, it's definitely possible then, and relevant. Once I hear from Kyle I'll update the text.

Reply:
It seems like there's code to fetch a patron's name from Clever login, yes. I haven't verified it works.

Reply:
I'm going to write "The Open eBooks client displays the patron's personal name to show who is logged in. This is important in an environment where devices are shared across a classroom." Let me know if it goes beyond that.

Reply:
There's a method to extract the full name but I dont see it used anywhere.
I don't think we even display the user's name to be honest.

Reply:
And yes, I believe we do wipe the user's name/info when we wipe the account details. So for example when we sign out.

Reply:
I bet that was a requirement at one point but we dropped the ball. We should either stop sending this information to the client or stop using it. I filed https://jira.nypl.org/browse/SIMPLY-2372 to cover this.